### ISSUE TITLE
Modernize configuration

### ISSUE THREAD

**Issue by zachdaniel** [zachdaniel/spandex#38]
We should follow modern patterns for configuration, and use a pattern whereby configuration is scoped to a module, such that there could be multiple, differently configured tracers.

**Comment by zachdaniel** [comment]
The current plan is to provide an `init` callback, as well as a `use` statement to configure your own tracer. Per the recommendations here: https://michal.muskala.eu/2017/07/30/configuring-elixir-libraries.html, the functions in `Spandex` will all take their entire configuration as opts. Then an implementor of the library can `use Spandex.Tracer`, and override any functions they'd like to override.

```
defmodule MyApp.Tracer do
  use Spandex.Tracer

  def init() do
    [
      service: :my_app,
      adapter: MyAdapter,
      ...
    ]
  end
end
```

This just boils down to providing those opts to all of the `Spandex` module's functions without having to specify them all the time. This empowers the most use cases possible in my opinion.

**Comment by zachdaniel** [comment]
Closed by #48. README is updated accordingly.
